src/number.js

```diff
@@ -17,7 +18,10 @@ export default function NumberSchema() {
       if (this.isType(value)) return value;
 
       let parsed = parseFloat(value);
-      if (this.isType(parsed)) return parsed;
+      if (this.isType(parsed) && isFinite(value)) return parsed;
+
+      if (typeof value === 'string' && value !== '' && this.isType(+value))
+        return +value;
```

Comment:
Sorry for the delay, I've been trying to think what the best option here is in terms of fixing number parsing more generally.

Rereading everything, it seems like `Number(value)` is the best bet (as you have here) and we remove the parseFloat case. What do you think?

Reply:
ActuallyI think it has a problem now that I've revisited it. I realised there is no test for whitespace characters.